### PR TITLE
Search with Tag feature added

### LINKED ISSUE #285: Search from tag optionality
Feature of searching posts through the tag. Use of tag can be achieved.
User will be shown the posts which the search he / she is searching with the tags.

Please assign this to me.

### PR DESCRIPTION
**Pull request by AmeyaJain-25**
Closes #285 
Now user can search from tags.
1. If tag is present, then that post will be seen while searching.
2. If fav checkbox is chekcked, then tag containing fav is seen.
3. If Searchbox is empty, then all posts are seen.
Screenshot from 2021-05-31 23-29-46

### CODE REVIEW

**Comment by sanjay270899** [comment]
@AmeyaJain-25 Please solve conflicts.

File: client/src/components/Posts/Posts.js [hunk 1]
```diff
@@ -45,6 +46,21 @@ const Posts = ({setCurrentId, setOpenCreatePost}) =>{
         }
     }
 
+    // function searchStringInArray (str, strArray) {
```
Comment (sanjay270899) [inline client/src/components/Posts/Posts.js:49]: If not needed can we remove it?

**Review by sanjay270899 — COMMENTED** [review]
@AmeyaJain-25

File: client/src/components/Posts/Posts.js (hunk 1 above)
Comment (AmeyaJain-25) [reply client/src/components/Posts/Posts.js:49]: Actually there were two types of search. But then it wasn't needed, so I commented.
Yes we can remove it.
I'll remove it and then push the code

**Comment by AmeyaJain-25** [comment]
@sanjay270899 
Made the required changes

**Pull request merged**